server/app/views/components/TextFormatter.java

```diff
@@ -38,6 +41,11 @@ public static ImmutableList<DomContent> formatText(
     return builder.build();
   }
 
+  /** Passes provided text through Markdown formatter without preserving empty lines or required indicators. */
+  public static ImmutableList<DomContent> formatText(String text) {
```

Comment:
praise: smart helper method! so much easier than copy/pasting those param comments. will you use this methods in the other places it's applicable as well?

Reply:
Yes! I was thinking I would do that in a separate PR so it's clear which changes are part of the fix for markdown rendering and which are a code cleanup. 

Also, sorry for the long delay! I've been wrestling with my local environment and fixing tests. I think I've finally figured out how to make PR. I'm going to close out this PR because it's from a fork and add you to a clean PR with the changes you recommended in the comment.